src/services/matrix.rs

```diff
@@ -72,47 +77,23 @@ pub mod matrix {
         pub send_to_thread: bool,
     }
 
-    pub async fn create_client(homeserver_url_str: String) -> Client {
+    pub async fn create_client(homeserver_url_str: &str) -> Client {
         info!("create client ");
         let homeserver_url =
             Url::parse(&homeserver_url_str).expect("Couldn't parse the homeserver URL");
-        let client = Client::new(homeserver_url).await.unwrap();
+        let client = Client::new(homeserver_url)
+            .await
+            .expect("can't handle new Client: create_client");
 
         client
     }
 
-    pub async fn login_and_sync(
-        client: &Client,
-        username: String,
-        password: String,
-    ) -> Result<String, String> {
-        info!("Logging");
-        let response = client
-            .login_username(&username, &password)
-            .initial_device_display_name("rust-sdk")
-            .send()
-            .await;
-
-        match response {
-            Ok(res) => {
-                info!("res: {:?}", res);
-                info!("Syncing");
-                client.sync_once(SyncSettings::default()).await.unwrap();
-
-                return Ok(String::from("Welcome"));
-            }
-            Err(err) => match err {
-                Error::Http(HttpError::Api(FromHttpResponseError::Server(ServerError::Known(
-                    matrix_sdk::RumaApiError::ClientApi(m),
-                )))) => return Err(m.message),
-                _ => return Err(String::from("An error have been ocurred!")),
-            },
-        }
-    }
-
     pub async fn join_room(client: &Client, room_id: &RoomId) {
```

Comment:
If we convert this function to return a Result then it won't be doing anything really meaningful and we could just ditch it

Reply:
Maybe we can leave it there for now to separate having the client code isolated, in the future we can extract all the client code and it will be easier if we have it in one place